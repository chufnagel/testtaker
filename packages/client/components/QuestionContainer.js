import PropTypes from "prop-types";
import React, { Component } from "react";

import Answer from "./Answer";
import Question from "./Question";

class QuestionContainer extends Component {
  render() {
    const { question, handleSelect } = this.props;
    return (
      <div className="card">
        <Question id={question.id} text={question.text} />
        {question.answers.map(answer => (
          <Answer
            questionId={question.id}
            choice={answer.choice}
            key={answer.choice}
            text={answer.content}
            onSelect={handleSelect}
          />
        ))}
      </div>
    )
  }
}
// const QuestionContainer = ({ question, handleSelect }) => {
//   return (
//     <div className="card">
//       <Question id={question.id} text={question.text} />
//       {question.answers.map(answer => (
//         <Answer
//           questionId={question.id}
//           choice={answer.choice}
//           key={answer.choice}
//           text={answer.content}
//           onSelect={handleSelect}
//         />
//       ))}
//     </div>
//   );
// };

export default QuestionContainer;
