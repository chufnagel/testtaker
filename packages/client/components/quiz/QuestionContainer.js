import PropTypes from "prop-types";

import Answer from "./Answer";
import Question from "../Question";

const QuestionContainer = ({ question, handleSelect }) => (
  <div className="card">
    <Question id={question.id} text={question.text} />
    <form>
      {question.answers.map(answer => (
        <Answer
          questionId={question.id}
          choice={answer.choice}
          key={answer.choice}
          text={answer.content}
          onSelect={handleSelect}
          checked={answer.choice === question.userAnswer}
        />
      ))}
    </form>
  </div>
);

QuestionContainer.propTypes = {
  question: PropTypes.shape({
    id: PropTypes.number,
    text: PropTypes.string,
    answers: PropTypes.arrayOf(PropTypes.object),
  }),
  handleSelect: PropTypes.func.isRequired,
};

QuestionContainer.defaultProps = {
  question: {
    id: 1,
    text: "Lorem ipsum",
    answers: [{ a: "Lorem ipsum" }],
  },
};

export default QuestionContainer;
