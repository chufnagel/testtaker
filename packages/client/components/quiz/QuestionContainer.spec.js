/* global beforeEach describe, expect, test */
import { shallow } from "enzyme";
import QuestionContainer from "./QuestionContainer";

// QuestionContainer.propTypes = {
//   question: PropTypes.shape({
//     id: PropTypes.string,
//     text: PropTypes.string,
//     answers: PropTypes.arrayOf(PropTypes.obj),
//   }),
//   handleSelect: PropTypes.func.isRequired,
// };

const mockFn = jest.fn();

describe("QuestionContainer", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<QuestionContainer handleSelect={mockFn} />);
  });
  test("It should render without crashing", () => {
    expect(wrapper.exists()).toEqual(true);
  });
  test("QuestionContainer matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
