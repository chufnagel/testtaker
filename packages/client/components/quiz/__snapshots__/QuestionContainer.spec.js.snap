// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionContainer QuestionContainer matches snapshot 1`] = `
<div
  className="card"
>
  <Question
    id={1}
    text="Lorem ipsum"
  />
  <form>
    <Answer
      checked={true}
      choice="A"
      key="A"
      onSelect={[MockFunction]}
      questionId={1}
      text="Lorem ipsum"
    />
  </form>
</div>
`;
